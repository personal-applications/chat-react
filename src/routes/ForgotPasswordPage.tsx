import useSWRMutation from "swr/mutation";
import ForgotPasswordForm, {
  ForgotPasswordFields,
} from "../forms/ForgotPasswordForm";
import { postData } from "../services/fetcher";

function ForgotPasswordPage() {
  const { trigger } = useSWRMutation<
    { message: string },
    Error,
    "/api/auth/forgot-password",
    ForgotPasswordFields
  >("/api/auth/forgot-password", postData);

  const onSubmit = async (data: ForgotPasswordFields) => {
    const { message } = await trigger(data);

    return { message };
  };

  return (
    <>
      <div className="h-screen w-screen bg-neutral pt-40">
        <div className="w-[600px] mx-auto">
          <h1 className="text-3xl font-semibold text-center mb-5">ChatHub</h1>
          <h1 className="text-bold text-center text-xl">Forgot Password</h1>
          <ForgotPasswordForm onSubmit={onSubmit} />
        </div>
      </div>
    </>
  );
}

export default ForgotPasswordPage;
