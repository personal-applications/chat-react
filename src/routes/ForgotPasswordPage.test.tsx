import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import useSWRMutation from "swr/mutation";
import { Mock } from "vitest";
import ForgotPasswordPage from "./ForgotPasswordPage";

vi.mock("swr/mutation", () => {
  return {
    default: vi.fn().mockReturnValue({}),
  };
});

test("should render the forgot password form", () => {
  const tree = render(
    <BrowserRouter>
      <ForgotPasswordPage />
    </BrowserRouter>,
  );

  expect(tree).toMatchSnapshot();
});

test("should trigger the forgot password mutation on form submission", async () => {
  const triggerMock = vi.fn().mockResolvedValue({ message: "Success" });
  (useSWRMutation as Mock).mockReturnValue({ trigger: triggerMock });

  render(
    <BrowserRouter>
      <ForgotPasswordPage />
    </BrowserRouter>,
  );

  const emailInput = screen.getByPlaceholderText("Email");
  const submitButton = screen.getByText("Forgot");

  await userEvent.type(emailInput, "test@example.com");
  await userEvent.click(submitButton);

  await waitFor(() => {
    expect(triggerMock).toHaveBeenCalledTimes(1);
    expect(triggerMock).toHaveBeenCalledWith({ email: "test@example.com" });
    expect(screen.getByText("Success")).toBeInTheDocument();
  });
});
