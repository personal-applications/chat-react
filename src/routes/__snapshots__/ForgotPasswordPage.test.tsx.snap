// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the forgot password form 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen w-screen bg-neutral pt-40"
      >
        <div
          class="w-[600px] mx-auto"
        >
          <h1
            class="text-3xl font-semibold text-center mb-5"
          >
            ChatHub
          </h1>
          <h1
            class="text-bold text-center text-xl"
          >
            Forgot Password
          </h1>
          <form
            class="mt-6 flex flex-col gap-y-6 rounded-md bg-base-100 p-8"
            novalidate=""
          >
            <div
              class="flex w-full flex-col"
            >
              <label
                class="input input-bordered flex items-center gap-2"
              >
                <input
                  class="grow"
                  name="email"
                  placeholder="Email"
                  type="email"
                />
              </label>
            </div>
            <div
              class="ml-auto"
            >
              <a
                class="link link-primary link-hover"
                href="/login"
              >
                Back to login
              </a>
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Forgot
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen w-screen bg-neutral pt-40"
    >
      <div
        class="w-[600px] mx-auto"
      >
        <h1
          class="text-3xl font-semibold text-center mb-5"
        >
          ChatHub
        </h1>
        <h1
          class="text-bold text-center text-xl"
        >
          Forgot Password
        </h1>
        <form
          class="mt-6 flex flex-col gap-y-6 rounded-md bg-base-100 p-8"
          novalidate=""
        >
          <div
            class="flex w-full flex-col"
          >
            <label
              class="input input-bordered flex items-center gap-2"
            >
              <input
                class="grow"
                name="email"
                placeholder="Email"
                type="email"
              />
            </label>
          </div>
          <div
            class="ml-auto"
          >
            <a
              class="link link-primary link-hover"
              href="/login"
            >
              Back to login
            </a>
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Forgot
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
